### ISSUE TITLE
[8.0.0.-beta2-rev1] Data from a merged cell is gone if its part is hidden - empty editor

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#6872]
### Description

Data from a merged cell is gone if its part is hidden. It is a regression from this issue https://github.com/handsontable/handsontable/issues/6871 - previously it was just a UI issue, in 8.0.0.-beta2 data is also gone from the editor.

### Steps to reproduce

1. Go to https://jsfiddle.net/xc7uaohL/
2. Open editor for the merged cell - tthere **should** be `B2` in the editor but it's blank
3. Type `test` in the merged cell - you **can** see `test` in the cell
4. Click the button below to log data in the console 
5. Show columns
6. Unmerge cells - the old data is visible - `B2`, not `test`
7. Click the button below to log data in the console - you've changed **C2**, not B2

If you at this point click the button, you'll get `B2` on its position and `test` on C2. Which also reflects if you unmerge the cells.

![beta2](https://user-images.githubusercontent.com/10757813/81783929-64adc280-94fc-11ea-98e7-c59c96316787.gif)

### Previously
1. Go to https://jsfiddle.net/xc7uaohL/
2. Open editor for the merged cell - there's `B2`
3. Type `test` in the merged cell -  you **cannot** see `test` in the cell
4. Click the button below to log data in the console
5. Show columns
6. Unmerge cells - the `NEW` data is visible - `test`
7. Click the button below to log data in the console - you've changed **B2**

![beta1](https://user-images.githubusercontent.com/10757813/81781924-45616600-94f9-11ea-9590-706c2950c3b3.gif)

### Demo

https://jsfiddle.net/handsoncode/8ffpsqt6/ 7.4.2
https://jsfiddle.net/4brxm2df/2/ 8.0.0.-beta2 from commit 929b22cff79f4b3d465a3d0149bbbd7c4acd0b34 (**last working**)
https://jsfiddle.net/4brxm2df/3/ 8.0.0.-beta2 from commit 81020af6ad04e8cd6b390c22e7b849a61a529286 (**first broken**)
https://jsfiddle.net/4brxm2df/4/ 8.0.0.-beta-rev4 (still broken)

### Related issues
* https://github.com/handsontable/handsontable/issues/6871

### Your environment
* Handsontable version: 8.0.0.-beta2 from commit 81020af6ad04e8cd6b390c22e7b849a61a529286
* Browser Name and version: Chrome 80
* Operating System: macOS Catalina

**Comment by AMBudnik** [comment]
Works the same if we hide right part (without visible data) https://jsfiddle.net/7a3cq2mt/
![Zrzut ekranu 2020-05-13 o 10 24 06](https://user-images.githubusercontent.com/10757813/81789178-e48b5b00-9503-11ea-9bae-42ff72410979.png)

**Comment by AMBudnik** [comment]
If we go to https://jsfiddle.net/xc7uaohL/ and click CTRL/CMD + Z to undo the `C2` is back.

**Comment by wszymanski** [comment]
May be related to #6873.

**Comment by aninde** [comment]
When we merge cells, the first of which is hidden, we don't see the content and we have an empty, merged cell. http://jsfiddle.net/aninde/ehs1obzq/1/ `8.0.0-beta2-rev10`. When the first row ceases to be hidden  A2 is displayed on the merged cells.

```
 hiddenRows: {
      rows:[1,2,3]
    },
 mergeCells: [
      {"row":1,"col":0,"rowspan":6,"colspan":1}
    ]
```
<img width="149" alt="Screenshot 2020-06-03 at 16 52 04" src="https://user-images.githubusercontent.com/30186818/83652587-151e5c00-a5bb-11ea-8823-f8acfb2b228d.png">

```
 hiddenRows: {
      rows:[2,3]
    },
 mergeCells: [
      {"row":1,"col":0,"rowspan":6,"colspan":1}
    ]
```
<img width="160" alt="Screenshot 2020-06-03 at 16 52 14" src="https://user-images.githubusercontent.com/30186818/83652630-20718780-a5bb-11ea-9d60-232a7778beb9.png">

**Comment by AMBudnik** [comment]
### Case 2

Steps
1. Go to https://jsfiddle.net/AMBudnik/kz98cq1e/ 8.0.0-beta2-rev3
2. Select **A1**
3. Hold <kbd>CMD/CTRL</kbd> and click **E2**

**Result**
You selected `A1:E3` instead of `A1:E2`.

![issue](https://user-images.githubusercontent.com/10757813/83735365-996eee80-a650-11ea-8946-b81c111733d0.gif)


The issue's also replicable in 8.0.0-beta2-rev10 https://jsfiddle.net/AMBudnik/k6robep7/1/

**Comment by wojciechczerniak** [comment]
> ### Case 2
> 
> Steps
> 
> 1. Go to https://jsfiddle.net/AMBudnik/kz98cq1e/ 8.0.0-beta2-rev3
> 2. Select **A1**
> 3. Hold CMD/CTRL and click **E2**
> 
> **Result**
> You selected `A1:E3` instead of `A1:E2`.
> 
> ![issue](https://user-images.githubusercontent.com/10757813/83735365-996eee80-a650-11ea-8946-b81c111733d0.gif)
> 
> The issue's also replicable in 8.0.0-beta2-rev10 https://jsfiddle.net/AMBudnik/k6robep7/1/

Exactly how it supposed to. This is the desired behavior. Check XL and GS.

**Comment by AMBudnik** [comment]
@wojciechczerniak is right. Works the same way in Google Sheets 🤔 but I did not change my mind when it comes to UX this looks odd.

**Comment by AMBudnik** [comment]
@wojciechczerniak what about this, let's call it **case 3**

Steps
1. Select both columns 
2. Use `Merge cells` from the context menu
3. Use <kbd>CTRL/CMD</kbd> + <kbd>Z</kbd> to undo

**Result**
![merge-show-undo](https://user-images.githubusercontent.com/10757813/83737415-4185b700-a653-11ea-97f1-02c1ce55e820.gif)

I would expect to get the same result as we get (cause we do not support undo-redo for hiding/showing) but with header `A` highlighted (not all of them).


Here's Google Sheets reproduction for comparison

![gs](https://user-images.githubusercontent.com/10757813/83737765-bb1da500-a653-11ea-8a5d-b8443b8b16f6.gif)

**Comment by wojciechczerniak** [comment]
True. Only A header should be selected. Is this a regression?

For undo/redo hidding column we should have a separate issue. This is out of scope

**Comment by AMBudnik** [comment]
> True. Only A header should be selected. Is this a regression?

Broken in 8.0.0-beta2-rev10 https://jsfiddle.net/AMBudnik/2t5sdgoz/ as well as 7.4.2 (and 6.2.2) https://jsfiddle.net/AMBudnik/bp0xrv1y/ - **not a regression**. I'll move it to a separate issue.

**Comment by wszymanski** [comment]
> When we merge cells, the first of which is hidden, we don't see the content and we have an empty, merged cell. [http://jsfiddle.net/aninde/ehs1obzq/1](http://jsfiddle.net/aninde/ehs1obzq/1/) `8.0.0-beta2-rev10`. When the first row ceases to be hidden A2 is displayed on the merged cells.

Probably related to the Handsontable 7+. Please check if extra information should be added to the https://github.com/handsontable/handsontable/issues/6871.

**Comment by AMBudnik** [comment]
@wszymanski @aninde 

> Probably related to the Handsontable 7+. Please check if extra information should be added to the #6871.

all - 8.0.0.beta2-rev6 http://jsfiddle.net/AMBudnik/mvntg5pb/
![6 2 2](https://user-images.githubusercontent.com/10757813/83741543-9f68cd80-a658-11ea-82d0-3feb70623c0c.png)


8.0.0.beta2-rev7 - rev10 http://jsfiddle.net/AMBudnik/2yq6zj91/
![Zrzut ekranu 2020-06-4 o 11 44 00](https://user-images.githubusercontent.com/10757813/83741629-ba3b4200-a658-11ea-8d27-c194596334ab.png)

We may say that it never displayed the value and before rev7 it was broken even worse.

**Comment by aninde** [comment]
@AMBudnik you're right, this issue (https://github.com/handsontable/handsontable/issues/6872#issuecomment-638743052) has already been partially reported here https://github.com/handsontable/handsontable/issues/6224#issuecomment-638241274.

**Comment by AMBudnik** [comment]
The main issue is **solved** in 8.0.0-beta2-rev11 https://jsfiddle.net/AMBudnik/ctjm0zhv/ it works the same way as it worked before the regression.

As **case 2** is intended, **case 3** is moved and the **case** with

> When we merge cells, the first of which is hidden, we don't see the content 

is reported in other issues we can close this topic. 

Thanks guys!
